server/app/services/rpc_client.rb

```diff
@@ -3,17 +3,17 @@
 class RpcClient
 
   class Error < StandardError
-    attr_accessor :code, :message, :backtrace
+    attr_reader :code
 
-    def initialize(code, message, backtrace = nil)
-      self.code = code
-      self.message = message
-      self.backtrace = backtrace
+    def initialize(code, message, remote_backtrace = nil)
+      @code = code
+      super(message)
+      set_backtrace(caller) unless backtrace
```

Comment:
Seems like the `def backtrace` approach gets even more complicated... is the previous `initialize -> set_backtrace(caller)` thing better?